wallet/tickets.go

```diff
@@ -76,15 +78,30 @@ func (w *Wallet) LiveTicketHashes(rpcClient *chain.RPCClient, includeImmature bo
 		return nil, err
 	}
 
-	for _, h := range stakeMgrTickets {
-		if sliceContainsHash(ticketHashes, h) {
+	for _, h := range ticketHashes {
+		ticketMap[h] = struct{}{}
+	}
+
+	promisesGetTxOut := make([]promiseGetTxOut, 0, len(stakeMgrTickets))
+	promisesGetRawTransaction := make([]promiseGetRawTransaction, 0, len(stakeMgrTickets))
+
+	// Get the raw transaction information from daemon and add
+	// any relevant tickets. The ticket output is always the
+	// zeroth output.
+	for i, h := range stakeMgrTickets {
+		_, exists := ticketMap[h]
+		if exists {
 			continue
 		}
 
-		// Get the raw transaction information from daemon and add
-		// any relevant tickets. The ticket output is always the
-		// zeroeth output.
-		spent, err := rpcClient.GetTxOut(&h, 0, true)
+		promisesGetTxOut = append(promisesGetTxOut, promiseGetTxOut{
+			result: rpcClient.GetTxOutAsync(&h, 0, true),
```

Comment:
Yeah I did that because I saw something weird (duplicate hashes in the result) when testing some changes but it was a mistake elsewhere.

Updated.